src/main/java/org/spongepowered/common/mixin/core/item/merchant/MixinITradeList.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.mixin.core.item.merchant;
+
+import net.minecraft.village.MerchantRecipeList;
+import org.spongepowered.api.item.merchant.TradeOffer;
+import org.spongepowered.api.item.merchant.generator.TradeOfferGenerator;
+import org.spongepowered.api.util.annotation.NonnullByDefault;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.common.item.merchant.SpongeTradeOfferBuilder;
+
+import java.util.List;
+import java.util.Random;
+
+@NonnullByDefault
+@Mixin(net.minecraft.entity.passive.EntityVillager.ITradeList.class)
+public abstract class MixinITradeList implements TradeOfferGenerator {
+
+    private static final Random RANDOM = new Random();
+
+    @Shadow
+    public abstract void modifyMerchantRecipeList(MerchantRecipeList recipeList, Random random);
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public void generate(final List<TradeOffer> tradeOffers) {
+        if (tradeOffers instanceof MerchantRecipeList) {
+            modifyMerchantRecipeList((MerchantRecipeList) tradeOffers, RANDOM);
+        } else {
+            final MerchantRecipeList list = new MerchantRecipeList();
+            for (final TradeOffer offer : tradeOffers) {
```

Comment:
addAll is a good idea, i haven't used it before because there have been some casts.
ctor is not possible since minecraft does not specify it anymore in MerchantRecipeList